src/PostgREST/App.hs

```diff
@@ -332,10 +333,10 @@ addFilter (path, flt) (Node rn forest) =
       where maybeNode = find ((name==).fst.snd.rootLabel) forst
 
 toSourceRelation :: Text -> Relation -> Maybe Relation
-toSourceRelation mt r@(Relation _ t _ ft _ _ rt _ _)
-  | mt == t = Just $ r {relTable=sourceSubqueryName}
-  | mt == ft = Just $ r {relFTable=sourceSubqueryName}
-  | Just mt == rt = Just $ r {relLTable=Just sourceSubqueryName}
+toSourceRelation mt r@(Relation t _ ft _ _ rt _ _)
```

Comment:
This function confuses me, and leads me into other confusing code as well. I went to see how it is used, and found it in `parseRequest` and wow that did not clear things up.

Maybe it would help to not pattern match into tiny variable names. I don't know what `mt` means, or the difference between `relFTable` and `relLTable` inside `Relation`. With more digging through I'll be able to piece it together but these names aren't helping.